dashboard/app/models/user_script.rb

```diff
@@ -39,9 +39,12 @@ def empty?
     started_at.nil? && completed_at.nil? && assigned_at.nil? && last_progress_at.nil?
   end
 
-  def check_plc_task_assignment_updating
-    if self.completed_at && self.script.pd
-      Plc::ScriptCompletionTask.check_for_script_completion(self)
+  def mark_script_completion_levels
+    if completed_at && script.pd
+      ScriptCompletion.all.
+        select{ |sc| sc.script_id == script.id.to_s }.
+        map(&:script_levels).flatten.
+        each{ |sl| user.track_level_progress_async(sl, ActivityConstants::BEST_PASS_RESULT, false) }
```

Comment:
Our style guide recommends `{...}` for chaining: https://github.com/bbatsov/ruby-style-guide#single-line-blocks

(It also recommends not chaining across multiple lines, but [we tend to do this in our codebase](https://github.com/code-dot-org/code-dot-org/blob/24343fad6bf1327fda31714ea13da766587d0a5e/dashboard/app/models/plc/script_completion_task.rb#L31-L35).)